frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+import {
+    lookupTableApi,
+    LookupTable,
+    ValueSet,
+    LookupTables,
+} from "../network/api/LookupTableApi";
+
+export const generateUseLookupTable =
+    <T extends unknown>(tableName: LookupTables) =>
+    () => {
+        return useLookupTable<T>(tableName);
+    };
+
+export async function GetLatestVersion(
+    tableName: LookupTables
+): Promise<number> {
+    let response;
+    try {
+        response = await axios(lookupTableApi.getTableList()).then(
+            (response) => response.data
+        );
+
+        let filteredBody: LookupTable[] = response.filter(
+            (tv: LookupTable) => tv.tableName === tableName && tv.isActive
+        );
+        if (filteredBody.length === 0) {
+            filteredBody = response.filter(
+                (tv: LookupTable) => tv.tableName === tableName
+            );
+        }
+        const table: LookupTable = filteredBody.sort(
+            (a: LookupTable, b: LookupTable) =>
+                b["tableVersion"] - a["tableVersion"]
+        )[0];
+
+        return table?.tableVersion;
+    } catch (error) {
+        return -1;
```

Comment:
@doug-s-nava added some better error handling. Still returning -1 because I have to return _something_. But the experience here tests out as: an error dialog pops up (toast) and no data displays on the page